sips/sip-024/sip-024-least-supertype-fix.md

```diff
@@ -0,0 +1,115 @@
+
+```clarity
+(list (tuple (a 1)) (tuple (b 1) (a 1)))
+```
+
+The Clarity runtime will treat each item of this list as if it only
+had the field `a`. This creates an issue for database reads and writes.
+On database reads, the Clarity database checks if the found type
+matches the expected type, and discovers a mismatch. This mismatch
+led to the DoS on 8 May 2023, and was fixed by converting the node
+crash into a transaction invalidation.
+
+However, transaction invalidation is _not_ sufficient as a long-term
+solution:
+
+1. Miners must be able to charge for these kinds of failures
+2. Contracts which do not directly rely on this behavior could still
+   receive buggy values because of the behavior. They should not
+   have storage failures because of buggy values they could not
+   control.
+
+# Specification
+
+The proposed changes to the Epoch 2.4 hard fork will do the following:
+
+* Add a value sanitization routine which eliminates any of these extra
+  fields from the in-memory representation of a Clarity value.
+* Invoke the sanitization routine on contract-call arguments and
+  return values.
+* Invoke the sanitization routine on database reads.
+* Invoke the sanitization routine during Clarity value constructors
+  which relied on the buggy type check behavior.
+
+This will preserve the existing type system behavior, but it will ensure
+that values constructed this way _match_ the expected type.
+
+# Related work
+The Stacks network has precedent for fixing consensus bugs through hard forks, some being released on 
+short timelines. 
+
+Other blockchains have also detected and fixed consensus critical bugs quickly. A prominent example of 
+this happened on Bitcoin, which had a bug that would allow the minting of an arbitrary amount of BTC 
+above the 21 million cap. A patched version was quickly released, and the network upgraded in a 
+matter of hours. 
+
+# Backwards Compatibility 
+Everyone who runs a 2.3 node will be able to run a Stacks 2.4 node 
+off of their existing chainstate. There are no changes to the chainstate database schemas in this SIP.
+
+Stacks 2.4 nodes will not interact with Stacks 2.1 nodes on the peer network after the 
+Bitcoin block activation height passes. In addition, Stacks 2.4 nodes
+will ignore block-commits from Stacks 2.3 nodes (as well as from nodes on prior versions). 
+Similar changes were made for Stacks 2.05 and Stacks 2.1 to ensure that the new network
+cleanly separates from stragglers still following the old rules.
+
+# Activation 
+The changes described in this SIP will ship in the same release as the changes described in SIP-022, which discusses
+a proposes a fix to the proof of transfer protocol.
+
+This release will ship 400 blocks prior to reward cycle 60, which is Bitcoin block height 791,651. 
+This gives stackers ample time (almost 3 days) to stack through the new contract. 
+
+The node software for Stacks 2.4 shall be merged to the `master` branch of the
```

Comment:
Based on the timing proposed of bitcoin block `791551`, it may be necessary to alter the wording here slightly. 
400 blocks before the start of the prepare phase for cycle 60 is likely > 3 days. 

I would suggest altering the wording from `activation` to `prepare` phase, or simply bumping the day +1 since that's likely to be required based on the values proposed in sip-022.